web/app/controllers/bloggs_controller.rb

```diff
@@ -12,17 +12,19 @@ def show
 
   def new
     @article = Blogg.new
+    authorize! :new, @article
```

Comment:
yeah rails has a crap ton of magic.  `authorize!` is part of the CanCan gem (library).  the exclamation point seems to be useless in this context, except to make it obvious that it's `authorize` is a method (http://stackoverflow.com/questions/612189/why-are-exclamation-marks-used-in-ruby-methods) but I just followed the example so I kept it in.

:new is the action of a resource-- rails generates resources (which are objects that follow CRUD [create, read, update, delete]).  resources have actions, which can be referenced with specific keywords, such as :new, :edit, :create, etc.  

then @article is obviously the article


As for the syntax, the symbols after the method call (authorize!) are the arguments.

authorize takes 2 + x arguments, where the first 2 are the action and the subject (the instance of a resource) 
https://github.com/ryanb/cancan/blob/4560928dc375f9b31de00381c98334eb6aabc4b9/lib/cancan/ability.rb